### DISCUSSION TITLE
Eigensolver executioner w/ cantilever proble (tensor_mechanics)?

### DISCUSSION THREAD

**Discussion by mangerij** [idaholab/moose#19077] [Q&A Modules: General] [answered]
I have been poking around and was wondering if there is an example in the repo for the standard cantilever problem (bending) but solved with the eigenvalue solver. 

Therefore, one would obtain the modes (frequencies and eigenvectors of displacements for disp_x disp_y disp_z).

thanks

**Comment by GiudGiud** [comment, marked as answer]
@bwspenc @jiangwen84

**Reply by lynnmunday** [reply]
There isn't an example of this.  The eigensolver doesn't work for mechanics.  I think @cbolisetti has looked into this and might have a branch that solves a problem like this.

**Reply by cbolisetti** [reply]
Yes, I don't think we cam solve this with the current moose or TM. @sveerara has created the necessary objects for this but it was never merged.  You can find tests in this branch 

https://github.com/sveerara/moose/tree/eigen

You'll have to compile with slepC to use this though.

**Reply by mangerij** [reply]
Very cool. I will poke around. Is there any reason why it wasn't merged?

**Reply by cbolisetti** [reply]
I am don't know unfortunately. May be @fdkong knows.

**Reply by fdkong** [reply]
The eigen solver is general, and it will work for general problems. 

But the issue we have is that physics kernels such as TM can not really form an exact matrix that is required by SLEPc.  Most of eigen sovlers are based on matrix. We have a matrix-free version, but that generate only one eigenvalue

**Reply by hugary1995** [reply]
Now that we have AD, shouldn't the matrix be accurate?
